security/oc_pki.c

```diff
@@ -64,6 +62,8 @@ pki_add_intermediate_cert(size_t device, int credid, const unsigned char *cert,
     c_size += 1;
   }
 
+  mbedtls_x509_crt int_ca;
+  mbedtls_x509_crt_init(&int_ca);
   int ret = mbedtls_x509_crt_parse(&int_ca, cert, c_size);
   if (ret < 0) {
     OC_ERR("could not parse intermediate cert: %d", ret);
```

Comment:
mbedtls_x509_crt_free(&int_ca);

Reply:
also `mbedtls_x509_crt_init(&id_cert_chain);` is not clean for error branch on line 81

Reply:
fixed